plugins/module_utils/network/sonic/config/mirroring/mirroring.py

```diff
@@ -0,0 +1,461 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The sonic_mirroring class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.cfg.base import (
+    ConfigBase,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+    remove_empties,
+)
+from ansible.module_utils.connection import ConnectionError
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.facts.facts import Facts
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    to_request,
+    edit_config
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils import (
+    update_states,
+    get_diff,
+    get_replaced_config,
+    get_normalize_interface_name
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.formatted_diff_utils import (
+    __DELETE_CONFIG_IF_NO_SUBCONFIG,
+    get_new_config,
+    get_formatted_config_diff
+)
+
+PATCH = 'patch'
+DELETE = 'delete'
+URL = 'data/sonic-mirror-session:sonic-mirror-session/MIRROR_SESSION/MIRROR_SESSION_LIST'
```

Comment:
The "facts" file for this resource module is using openconfig. Unless there is a good reason not to, this file should also use it. The main difference in code here would be the conversion of the argspec "_" to the Opencofing "-" for applicable options. (The REST API structure below the top level looks the same, but please verify that and correct the path specifications as needed.)

```suggestion
URL = 'data/openconfig-mirror-ext:mirror/sessions/session'
```

Reply:
This looks like it wouldn't require much code change, and the "type" option is not really needed. It will be less confusing to users and future maintainers if we use the same Yang basis for both files.